src/helpers/ring.rs

```diff
@@ -0,0 +1,201 @@
+//!
+//! This module contains implementations and traits that enable MPC helpers to communicate with
+//! each other. In order for helpers to send messages, they need to know the destination. In some
+//! cases this might be the exact address of helper host/instance (for example IP address), but
+//! in many situations MPC helpers orchestrated into a "ring" - every helper instance has a peer
+//! on the right side and on the left side. They simply need to be able to send messages to the
+//! corresponding helper without needing to know the exact location - this is what this module
+//! enables MPC helper service to do.
+//!
+use crate::helpers::error::Error;
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::fmt::Debug;
+
+/// Trait for messages sent between helpers
+pub trait Message: Debug + Send + Serialize + DeserializeOwned + 'static {}
+
+impl<T> Message for T where T: Debug + Send + Serialize + DeserializeOwned + 'static {}
+
+/// Destination. Currently we only support Left and Right, but we could support the exact address
+/// too
```

Comment:
Makes sense. I can imagine other cases where we could have a KING or some such thing (e.g. when all the helpers need to send data to a single place to be revealed).